### DISCUSSION TITLE
Maybe `save_shape()` ?

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#223] [Ideas]
I know this strays a bit from the priorities of making Processing 4 things work but...

This could potentially be handy. Here some crazy ideas that passed my mind:

- `save_shape()` could initially just serialize a Py5Shape object (and `load_shape()` would have to be modified to identify and load it back, maybe using a `.py5shape` extension);
- In the future it could be a simple way of exporting SVG or even OBJ vector data... (that `load_shape()` can already load).

**Comment by hx2A** [comment]
This would be a great feature but no Java object can be pickled. Also, Processing links the PShape objects to the parent PApplet that created them. Loading a Py5Shape object from a pickle file would result in crazy errors, as that parent link would no longer be valid.

However, this is a useful feature. The way to implement this would be for Processing4 to add a save method to PShape. The relevant data could be saved to a file and then loaded back. Adding the feature to Processing would also add the feature to py5.

**Reply by villares** [reply]
I see... there is a whole Java machinery for serialization, and Jython used to tap on that I think, but I wouldn't want to go near it. :laughing:

**Reply by hx2A** [reply]
Theoretically py5 could do that also because you can write hooks for pickling of particular object types. It would be a lot of work though. Adding a save feature to Processing might be easier, and would certainly be a more robust approach.
